snuba/subscriptions/data.py

```diff
@@ -150,16 +151,26 @@ def build_request(
             Language.LEGACY,
         )
         extra_conditions: Sequence[Condition] = []
-        if offset is not None:
-            extra_conditions = [[["ifNull", ["offset", 0]], "<=", offset]]
+
+        # This check is necessary because datasets like the `sessions` dataset does not contain a
+        # column `offset`
+        if (
+            offset is not None
+            and "offset" in dataset.get_default_entity().get_data_model()
+        ):
+            extra_conditions = [[["ifnull", ["offset", 0]], "<=", offset]]
+
+        build_request_dict = {
+            "project": self.project_id,
+            "conditions": [*self.conditions, *extra_conditions],
+            "aggregations": self.aggregations,
+            "from_date": (timestamp - self.time_window).isoformat(),
+            "to_date": timestamp.isoformat(),
+        }
+        if self.organization:
```

Comment:
@evanh if I remove this I get a `JsonValidationError` for entities that are not sessions